strawberry/sanic/utils.py

```diff
@@ -0,0 +1,35 @@
+from io import BytesIO
+from typing import Dict, List, Union
+
+from sanic.request import Request
+
+
+def convert_request_to_files_dict(request: Request) -> dict:
+    """
+    request.files has the following format, even if only a single file is uploaded:
+
+    {
+        'textFile': [
+            sanic.request.File(
+                type='text/plain',
+                body=b'strawberry',
+                name='textFile.txt'
+            )
+        ]
+    }
+
+    Note that the dictionary entries are lists.
+    """
+    request_files: dict = request.files
+    files_dict: Dict[str, Union[BytesIO, List[BytesIO]]] = {}
+
+    for field_name, file_list in request_files.items():
+        if len(file_list) == 1:
```

Comment:
Can the list ever be empty?

Reply:
The `reques_files.items()` list probably can. I will add a test for it. In case you were referring to the `file_list`, this one I probably can't be empty.

Reply:
I added a test that coveres missing files in multipart requests. (thats what would cause `request_files.items()` to be empty.

I could not properly cover the `else` part of the util, since this would require Strawberrys `replace_placeholders*` functions to support file lists (#766). As mentioned in the PRs description I would like to address #766 in a separate PR.

Reply:
Can you add a comment to the `else` as a little TODO that refers to #766?

Reply:
Or open an issue to track this once this PR is merged

Reply:
I added a TODO referring to #766 :+1: